src/components/Card.js

```diff
@@ -1,61 +1,135 @@
 export default class Card {
-  constructor(cardData, cardSelector, handleCardClick) {
+  constructor(
+    {
+      cardData,
+      cardSelector,
+      handlePreviewImage,
+      userId,
+      handleLikeButton,
+      handleDeleteButton,
+    },
+    myId
+  ) {
     this._name = cardData.name;
     this._link = cardData.link;
+    this._likes = cardData.likes;
+    this._ownerId = cardData.owner._id;
+    this._myId = myId;
+    this._id = cardData._id;
+    this._userId = userId; // not using this value as of now
     this._cardSelector = cardSelector;
-    this._handleCardClick = handleCardClick;
+    this._handlePreviewImage = handlePreviewImage;
+    this._handleDeleteButton = handleDeleteButton;
+    this._handleLikeButton = handleLikeButton;
   }
 
-  _getTemplate() {
-    const cardElement = document
-      .querySelector(this._cardSelector)
-      .content.querySelector(".card")
-      .cloneNode(true);
+  _handleDeleteButtonClick() {
+    this._handleDeleteButton(this._id);
+  }
 
-    return cardElement;
+  getId() {
+    return this._id;
+  }
+
+  isLiked() {
+    return this._likes.some((like) => like._id === this._myId);
+  }
+
+  setLikes(likes) {
+    console.log(likes);
+    this._likes = likes;
+    this._renderLikes();
+  }
+
+  _renderLikes() {
+    this._cardLikesCountElement =
+      this._cardElement.querySelector(".card__likes-count");
```

Comment:
```js

    this._cardLikesCountElement =
      this._cardElement.querySelector(".card__likes-count");
```
**Could be improved**

Please, find it only once in line 121



All elements that are found every time you call a method should be defined as **class fields** in the constructor, `_setEventListeners` (or in `generateCard`) only **1 time** not to waste resources on searching them again and again.

`Class fields` (the so called **class variables** with `this`) are accessible inside any method of the class. You shouldn’t pass **class fields** into the call of any method because you can use class fields inside all methods.
There can be `this._inputs`, `this._submitButton`, `this._cardImage`, `this._form` and so on.

Reply:
Changed.